agent_pool.go

```diff
@@ -0,0 +1,190 @@
+package tfe
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"net/url"
+)
+
+// Compile-time proof of interface implementation.
+var _ AgentPools = (*agentPools)(nil)
+
+// AgentPools describes all the agent pool related methods that the Terraform
+// Enterprise API supports.
+//
+// TFE API docs: https://www.terraform.io/docs/cloud/api/agents.html
+type AgentPools interface {
+	// List all the agent pools of the given organization.
+	List(ctx context.Context, organization string, options AgentPoolListOptions) (*AgentPoolList, error)
+
+	// Create a new agent pool with the given options.
+	Create(ctx context.Context, organization string, options AgentPoolCreateOptions) (*AgentPool, error)
+
+	// Read a agent pool by its ID.
+	Read(ctx context.Context, agentPoolID string) (*AgentPool, error)
```

Comment:
do you think it is worth adding support for the `sort` and `includes` params on `List` and the `includes` param on `Read`?
edit: the more i think about it, probably not right now. this is mostly to support the provider work and those things are necessary for that and can be added later

Reply:
Agreed